prometheus-server-metrics/src/main/scala/org/http4s/server/prometheus/PrometheusMetrics.scala

```diff
@@ -101,19 +102,44 @@ object PrometheusMetrics {
       now <- Sync[F].delay(System.nanoTime)
       _ <- emptyResponseHandler.traverse_(status =>
         Sync[F].delay {
-          serviceMetrics.requestDuration
+          serviceMetrics.tag.fold(
+            serviceMetrics.requestDuration
             .labels(reportMethod(m), ServingPhase.report(ServingPhase.HeaderPhase))
             .observe(SimpleTimer.elapsedSecondsFromNanos(start, headerTime))
+          )(tag => 
+            serviceMetrics.requestDuration
+            .labels(reportMethod(m), ServingPhase.report(ServingPhase.HeaderPhase), tag)
```

Comment:
If `tags` is a `Map` (see suggestion below), can you just call it `tags.values: _*` here and get rid of some of the duplication?

Even as one, can you *gulp* use that horrid option2Iterable implicit and get the same result?